Move to GH Actions for CI

bosschaert opened the pull request:
Add build GH workflow

trieloff commented:
I thought we'd run the semantic release process in GH actions too and remove the CircleCI config altogether

bosschaert commented:
@trieloff I've removed the CircleCI config (the CircleCI app complains about it being absent now) and added a semver github action that only runs on main.
